glean-core/src/metrics/uuid.rs

```diff
@@ -54,6 +55,28 @@ impl UuidMetric {
         glean.storage().record(glean, &self.meta, &value)
     }
 
+    /// Sets to the specified value, from a string.
+    ///
+    /// This should only be used from FFI. When calling directly from Rust, it
+    /// is better to use `set`.
+    ///
+    /// # Arguments
+    ///
+    /// * `glean` - The Glean instance this metric belongs to.
+    /// * `value` - The UUID to set the metric to.
+    pub fn set_from_str(&self, glean: &Glean, value: &str) {
```

Comment:
I would like @badboy 's eyes on this. Maybe a better solution would be to preserve the strength of the typing of the metric API but open up a way to record errors from outside. (FOG would use such a capability since its errors might not be on the same process as the Glean singleton)

Reply:
Yeah -- I thought we had a bug for that but I can't find it now.  It turns out to be tricky in a *generic* way that would work for all the metric types (without a new FFI for each of them).  Maybe that's the cost we are willing to pay / would be easier when uniffi is viable etc...

Reply:
We have [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1589675) on a general refactor of the error reporting system, I also thought we had something more directly for exposing that over FFI.

Anyway, this probably requires a bit more work than we can afford right now to make this happen for the UUID type.

Reply:
@badboy: So you are saying we go forward with this (slightly) suboptimal solution now, or hold off until we have the time to do error reporting from the language bindings?

Reply:
Go forward with this solution for now, UUIDs are already used and if this catches a potential bug in Python (and m-c?) code good.

Reply:
It's not the worst in the world to have it be `pub` in `glean-core`. That's specifically not an API crate after all.